spanner/integration_test.go

```diff
@@ -2083,150 +2096,162 @@ func TestIntegration_BasicTypes(t *testing.T) {
+		{col: "String", val: nil, wantWithDefaultConfig: NullString{}, wantWithNumber: NullString{}},
+		{col: "StringArray", val: nil, wantWithDefaultConfig: []NullString(nil), wantWithNumber: []NullString(nil)},
+		{col: "Bytes", val: nil, wantWithDefaultConfig: []byte(nil), wantWithNumber: []byte(nil)},
+		{col: "BytesArray", val: nil, wantWithDefaultConfig: [][]byte(nil), wantWithNumber: [][]byte(nil)},
+		{col: "Int64a", val: nil, wantWithDefaultConfig: NullInt64{}, wantWithNumber: NullInt64{}},
+		{col: "Int64Array", val: nil, wantWithDefaultConfig: []NullInt64(nil), wantWithNumber: []NullInt64(nil)},
+		{col: "Bool", val: nil, wantWithDefaultConfig: NullBool{}, wantWithNumber: NullBool{}},
+		{col: "BoolArray", val: nil, wantWithDefaultConfig: []NullBool(nil), wantWithNumber: []NullBool(nil)},
+		{col: "Float64", val: nil, wantWithDefaultConfig: NullFloat64{}, wantWithNumber: NullFloat64{}},
+		{col: "Float64Array", val: nil, wantWithDefaultConfig: []NullFloat64(nil), wantWithNumber: []NullFloat64(nil)},
+		{col: "Numeric", val: nil, wantWithDefaultConfig: NullNumeric{}, wantWithNumber: NullNumeric{}},
+		{col: "NumericArray", val: nil, wantWithDefaultConfig: []NullNumeric(nil), wantWithNumber: []NullNumeric(nil)},
+		{col: "JSON", val: nil, wantWithDefaultConfig: NullJSON{}, wantWithNumber: NullJSON{}},
+		{col: "JSONArray", val: nil, wantWithDefaultConfig: []NullJSON(nil), wantWithNumber: []NullJSON(nil)},
 	}
 
 	// See https://github.com/GoogleCloudPlatform/cloud-spanner-emulator/issues/31
 	if !isEmulatorEnvSet() {
 		tests = append(tests, []struct {
-			col  string
-			val  interface{}
-			want interface{}
+			col                   string
+			val                   interface{}
+			wantWithDefaultConfig interface{}
+			wantWithNumber        interface{}
 		}{
-			{col: "Date", val: nil, want: NullDate{}},
-			{col: "Timestamp", val: nil, want: NullTime{}},
+			{col: "Date", val: nil, wantWithDefaultConfig: NullDate{}, wantWithNumber: NullDate{}},
+			{col: "Timestamp", val: nil, wantWithDefaultConfig: NullTime{}, wantWithNumber: NullTime{}},
 		}...)
 	}
 
-	// Write rows into table first using DML.
-	statements := make([]Statement, 0)
-	for i, test := range tests {
-		stmt := NewStatement(fmt.Sprintf("INSERT INTO Types (RowId, `%s`) VALUES (@id, @value)", test.col))
-		// Note: We are not setting the parameter type here to ensure that it
-		// can be automatically recognized when it is actually needed.
-		stmt.Params["id"] = i
-		stmt.Params["value"] = test.val
-		statements = append(statements, stmt)
-	}
-	_, err := client.ReadWriteTransaction(ctx, func(ctx context.Context, tx *ReadWriteTransaction) error {
-		rowCounts, err := tx.BatchUpdate(ctx, statements)
-		if err != nil {
-			return err
+	for idx := 0; idx < 2; idx++ {
+		if idx == 1 {
+			UseNumberWithJSONDecoderEncoder()
+			defer testSetJSONProviderNumberConfig(false)
```

Comment:
nit: can we change this into
1. First get the current value of `UseNumberWithJSONDecoderEncoder`
2. A loop that loops through two boolean value true and false
3. Then calls `UseNumberWithJSONDecoderEncoder(boolValue)`
4. Then after the loop set the `UseNumberWithJSONDecoderEncoder` to the original value.

Reply:
Also: Are we actually using this to check whether we get the expected value? In other words: I don't see any tests that are conditional on whether `idx == 1` or `idx == 0`, and that has a different expected outcome depending on that.

Reply:
There is condition on whether idx is 0/1, https://github.com/googleapis/google-cloud-go/pull/9054/files#diff-3303adad40b1fa3c6b0a92ea1f4ad28c32bc84c52000e7a5e9fc8b37e05eaa85R2226-R2229